### PR TITLE
Add build targets to build and view API docs

### LINKED ISSUE #1911: Generate scaladoc of unstable
As we discussed today, we are lacking internal developer's documentation. Although the methods in the new code are usually well-documented, there is no good overview of each package. Additionally, the documentation is written for someone who would maintain the code, not use the code as an API. That is, there is an implicit assumption that the developers of a package have to read its codebase, even if they do not want to change it.

We should improve this situation. The most economic way is probably to write the developer's documentation in javadoc itself. We can draw some inspiration from Java's javadoc, e.g., [collections](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html).

The next question is how we should publish the api doc. We can add a target to `sbt`. In my experience, however, developers would not call this target just to read the docs. IMO, a more effective way is to publish the javadoc of unstable somewhere, e.g., at github pages. What do you think @shonfeder?

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #1911

## Context

We've agreed as a team that we should invest in more thorough API documentation.
Improving the ease with which we can build and view the docs is important for
this.

## Changes

- Add sbt task `browseApiDocs` to build the docs, then open the `index.html` in
  the default browser (on MacOS and Linux, or print the file location on Windows).
- Add makefile task to run the task, and start watching and autobuilding the
  docs on file updates.
- Add a section on documentation to `CONTRIBUTING.md`

## Reviewing

It would be helpful if this could be tried on a Mac to ensure that the macos
case works as expected.

-------

- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -283,6 +283,26 @@ writeup][writeup]
 For installation and configuration in vanilla emacs, see
 https://scalameta.org/metals/docs/editors/emacs.html
 
+## Documentation
+
+We maintain two main kinds of documentation: tutorials and manuals meant as user
+documentation and API documentation meant for developers.
```
Comment (thpani) [inline CONTRIBUTING.md:289]: Should we also mention RFCs/ADRs (which seem to fall somewhere in between)?

**Review by thpani — APPROVED** [review]
Confirmed working on macOS.

One question though: when the root view opens, I see `zio` as one of the packages? Any ideas why it ends up there?
Screen Shot 2022-07-29 at 14 25 30

**Comment by shonfeder** [comment]
Yeah, I'm not sure why ZIO is showing up there! I'll investigate, but will probably address in a separate PR, as it's not likely related to these changes.

File: CONTRIBUTING.md (hunk 1 above)
Comment (shonfeder) [reply CONTRIBUTING.md:289]: Ah yeah. Good idea.

Comment (shonfeder) [reply CONTRIBUTING.md]: how do you find https://github.com/informalsystems/apalache/pull/2026/commits/1a4b77c1623bb4cf480766d4f477d0cf5619db1e ?

Comment (thpani) [reply CONTRIBUTING.md]: Excellent, just a word fix in the comment below.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -283,6 +283,26 @@ writeup][writeup]
 For installation and configuration in vanilla emacs, see
 https://scalameta.org/metals/docs/editors/emacs.html
 
+## Documentation
+
+We maintain three principle kinds of documentation:
+
+- Tutorials and manuals meant as general user documentation can be found in [./docs/src](./docs/src).
+- ADRs and RFCs documenting design and technical decisions are meant for developers
+  and expert users and be found in [./docs/src/adr](./docs/src/adr)
```
Comment (thpani) [inline CONTRIBUTING.md:292]: I think this needs either a modal verb or declination?

```suggestion
- ADRs and RFCs documenting design and technical decisions are meant for developers
  and expert users and can be found in [./docs/src/adr](./docs/src/adr)
```

```suggestion
- ADRs and RFCs documenting design and technical decisions are meant for developers
  and expert users and are found in [./docs/src/adr](./docs/src/adr)
```

**Pull request merged**